### PR TITLE
Fix compiler warnings caused by enabling ipv6

### PR DESCRIPTION
**Pull request by mateuszkobak**
## Description

Fixing compiler warnings introduced by https://github.com/phoenix-rtos/phoenix-rtos-lwip/pull/93

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,7 +29,7 @@ ifeq (${LWIP_G3_BUILD}, yes)
 include g3/Makefile
 endif
 
-CFLAGS += -Wundef -Iinclude -Ilib-lwip/src/include -I"$(LWIPOPTS_DIR)"
+CFLAGS += -Wundef -Wno-char-subscripts -Iinclude -Ilib-lwip/src/include -I"$(LWIPOPTS_DIR)"
```
Comment (anglov) [inline Makefile:32]: Big nope. You disable this warning globally, also for our port part.

**Review by anglov — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (mateuszkobak) [reply Makefile]: Moved this to `LOCAL_CFLAGS`

File: drivers/pktmem.c [hunk 2]
```diff
@@ -19,7 +19,6 @@
 
 
 #define CACHE_LINE_SIZE 64
-#define PAGE_SIZE 4096
```
[the remark this answers is not in the document]
Comment (mateuszkobak) [reply drivers/pktmem.c]: Leaving this issue as it was. Respective warning shall remain present.

File: lib-lwip
[the remark this answers is not in the document]
Comment (mateuszkobak) [reply lib-lwip:1]: No changes in lib-lwip are needed now

**Review by anglov — APPROVED** [review]
LGTM but added @nalajcie and @badochov

File: Makefile [hunk 3]
```diff
@@ -33,6 +33,8 @@ CFLAGS += -Wundef -Iinclude -Ilib-lwip/src/include -I"$(LWIPOPTS_DIR)"
 
 NAME := lwip-core
 SRCS := $(LWIP_SRCS)
+# Disabling warnings from lib-lwip, as their code does not comply with these rules for now.
+LOCAL_CFLAGS += -Wno-char-subscripts -Wno-format-zero-length
```
Comment (badochov) [inline Makefile:37]: `-Wno-format-zero-length` arrises only in one place due to missing assertion description, IMO better fix would be to add this description

Comment (anglov) [reply Makefile:37]: I have mixed feelings. I don't want to change lib-lwip to much as it's ported code which aren't fixed upstream. On the other hand it's easy, one place change...

Comment (badochov) [reply Makefile:37]: Whatever you decide is fine with me

**Comment by nalajcie** [comment]
maybe pin these commits to the original JIRA ID with enabling IPv6?

Other than that - LGTM.

**Comment by mateuszkobak** [comment]
@nalajcie, done

**Comment by nalajcie** [comment]
nitpicking: `silent lib-lwip warnings` -> `silence` (verb, not adjective)

**Review by nalajcie — APPROVED** [review]

**Pull request merged**